testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestSparkCompatibility.java

```diff
@@ -0,0 +1,175 @@
+import static io.trino.tests.TestGroups.HIVE_SPARK_BUCKETING;
+import static io.trino.tests.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.utils.QueryExecutors.onHive;
+import static io.trino.tests.utils.QueryExecutors.onPresto;
+import static io.trino.tests.utils.QueryExecutors.onSpark;
+import static java.lang.String.format;
+
+public class TestSparkCompatibility
+        extends ProductTest
+{
+    // see spark-defaults.conf
+    private static final String PRESTO_CATALOG = "hive";
+
+    @Test(groups = {HIVE_SPARK_BUCKETING, PROFILE_SPECIFIC_TESTS})
+    public void testPrestoReadingTableCreatedByNativeSpark()
+    {
+        String baseTableName = "test_presto_reading_spark_native_buckets";
+
+        String sparkTableDefinition =
+                "CREATE TABLE %s (" +
+                        "  _string STRING" +
+                        ", _bigint BIGINT" +
+                        ", _integer INTEGER" +
+                        ", _real REAL" +
+                        ", _double DOUBLE" +
+                        ", _boolean BOOLEAN" +
+                        ") USING ORC" +
+                        " CLUSTERED BY (_string) INTO 4 BUCKETS";
+        onSpark().executeQuery(format(sparkTableDefinition, baseTableName));
+
+        String values = "VALUES " +
+                "('one', 1000000000000000, 1000000000, 10000000.123, 100000000000.123, true)" +
+                ", ('two', -1000000000000000, -1000000000, -10000000.123, -100000000000.123, false)" +
+                ", ('three', 2000000000000000, 2000000000, 20000000.123, 200000000000.123, true)" +
+                ", ('four', -2000000000000000, -2000000000, -20000000.123, -200000000000.123, false)";
+        String insert = format("INSERT INTO %s %s", baseTableName, values);
+        onSpark().executeQuery(insert);
+
+        Row row1 = row("one", 1000000000000000L, 1000000000, 10000000.123F, 100000000000.123, true);
+        Row row2 = row("two", -1000000000000000L, -1000000000, -10000000.123F, -100000000000.123, false);
+        Row row3 = row("three", 2000000000000000L, 2000000000, 20000000.123F, 200000000000.123, true);
+        Row row4 = row("four", -2000000000000000L, -2000000000, -20000000.123F, -200000000000.123, false);
+
+        String startOfSelect = "SELECT _string, _bigint, _integer, _real, _double, _boolean";
+        QueryResult sparkSelect = onSpark().executeQuery(format("%s FROM %s", startOfSelect, baseTableName));
+        assertThat(sparkSelect).containsOnly(row1, row2, row3, row4);
+
+        QueryResult prestoSelect = onPresto().executeQuery(format("%s FROM %s", startOfSelect, prestoTableName(baseTableName)));
+        assertThat(prestoSelect).containsOnly(row1, row2, row3, row4);
+
+        onSpark().executeQuery("DROP TABLE " + baseTableName);
+    }
+
+    @Test(groups = {HIVE_SPARK_BUCKETING, PROFILE_SPECIFIC_TESTS})
+    public void testPrestoReadingTableCreatedByHiveCompliantSpark()
```

Comment:
Add some commentary what "native spark", "hive compliant spark" mean.

Also, if "hive compliant spark" cannot reproduce the situation without our help (no bucketing support, doesn't set the required table properties), i am not sure if this test models a real-life scenario, does it?

Reply:
I'm not sure about the real-life scenario argument - it's quite possible that a user will have a Hive-compliant table created in Spark, which up until was also accessed by Hive, and would like to read it through Trino. I think it's better to have a tested scenario for that case, if only to know what to do.

Reply:
I forgot that this table wouldn't have bucketing applied by Spark. I agree this scenario seems "hacky", with bucketing and Spark table property being added manually. Would you like to remove it?